sklearn/inspection/_friedmans_h.py

```diff
@@ -0,0 +1,217 @@
+"""Friedman and Popescu's H-squared statistics"""
+
+import itertools
+
+import numpy as np
+
+from ..utils import Bunch, _get_column_indices, _safe_assign, _safe_indexing
+from ..utils.random import sample_without_replacement
+from ..utils.validation import _check_sample_weight, check_is_fitted
+
+
+def _calculate_pd_over_data(estimator, X, feature_indices, sample_weight=None):
+    """Calculates partial dependence over the data distribution.
+
+    It returns a 1D or 2D numpy array of the same length as X.
+    """
+
+    # Select grid columns and remove duplicates (will compensate below)
+    grid = _safe_indexing(X, feature_indices, axis=1)
+
+    # Unfortunately, the next line does not work in all cases, especially not in
+    # the important case of discrete pandas Dataframes.
+    try:
+        compressed = True
+        grid, ix_reconstruct = np.unique(grid, return_inverse=True, axis=0)
+    except:
```

Comment:
```suggestion
    except ExpectedErrorFrom_np_unique as e:
```
Put as `ExpectedErrorFrom_np_unique` the exact error that `np.unique` raises, e.g. `TypeError` or `ValueError`.

Reply:
Ahhh